### PR TITLE
[RISCV] Add scheduling model for Syntacore SCR7

### PR DESCRIPTION
**Pull request by asi-sc**
Syntacore SCR7 is rv64imafdcv_zba_zbb_zbc_zbs_zkn.
Scheduling model for RVV will be added later.
Overview: https://syntacore.com/products/scr7

### CODE REVIEW

**Review by michaelmaitland — APPROVED** [review]
Don't know much about the micro-architecture specific changes, but everything else LGTM.

File: llvm/lib/Target/RISCV/RISCVSchedSyntacoreSCR7.td [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+
+// This file covers scheduling model for rv64imafdc_zba_zbb_zbc_zbs
```
Comment (michaelmaitland) [inline llvm/lib/Target/RISCV/RISCVSchedSyntacoreSCR7.td:11]: According to 108406 I think this processor is rv64imafdcv_zba_zbb_zbc_zbs_zkn. Are we missing `v` and `zkn` here?

**Review by el-sc — APPROVED** [review]

File: llvm/lib/Target/RISCV/RISCVSchedSyntacoreSCR7.td (hunk 1 above)
Comment (michaelmaitland) [reply llvm/lib/Target/RISCV/RISCVSchedSyntacoreSCR7.td:11]: I think it would be good to add a FIXME here to add `v` and `zkn` here.

Comment (asi-sc) [reply llvm/lib/Target/RISCV/RISCVSchedSyntacoreSCR7.td]: addressed

**Pull request merged**